src/timeout.ts

```diff
@@ -259,14 +268,26 @@ export class CSOTTimeoutContext extends TimeoutContext {
     const { remainingTimeMS } = this;
     if (!Number.isFinite(remainingTimeMS)) return null;
     if (remainingTimeMS > 0) return Timeout.expires(remainingTimeMS);
-    throw new MongoOperationTimeoutError('Timed out before socket write');
+    throw new MongoOperationTimeoutError(`Timed out before socket write after ${this.timeoutMS}ms`);
   }
 
   get timeoutForSocketRead(): Timeout | null {
     const { remainingTimeMS } = this;
     if (!Number.isFinite(remainingTimeMS)) return null;
     if (remainingTimeMS > 0) return Timeout.expires(remainingTimeMS);
-    throw new MongoOperationTimeoutError('Timed out before socket read');
+    throw new MongoOperationTimeoutError(`Timed out before socket read after ${this.timeoutMS}ms`);
+  }
+
+  refresh(): void {
+    this.start = Math.trunc(performance.now());
+    this.minRoundTripTime = 0;
+    this._serverSelectionTimeout?.clear();
+    this._connectionCheckoutTimeout?.clear();
+  }
```

Comment:
Added refresh method to accommodate closing of cursors as well as `iteration` mode

Reply:
Seems fine, may need the original for debugging, can defer.

Reply:
Should I add the tracking of the previous start times here or should we just file a ticket?

Reply:
Let's bring it back when we need it, we shouldn't try to maintain something we don't use. The time's aren't relevant to a wall clock so it would really only be for us to debug and this could locally be reintroduced easily if that came up.